File: front-end/src/application/repository/vesselRepository.ts
import { Vessel } from "../../domain/entity/vessel";
import { Either } from "../../utils/ull-either";
import { DataError } from "../../utils/ull-data-error";
import { UllUUID } from "../../utils/ull-uuid";

export interface VesselRepository {
    save: (vessel: Vessel) => Promise<Either<DataError, Vessel>>;
    
    find: (id: UllUUID) => Promise<Either<DataError, Vessel>>;
    
    findAll: (pageRequest: any) => Promise<Either<DataError, Vessel[]>>;
    
    update: (id: UllUUID, vessel: Vessel) => Promise<Either<DataError, Vessel>>;
    
    delete: (id: UllUUID) => Promise<Either<DataError, Vessel>>;
    
    deleteAll: () => Promise<Either<DataError, Vessel[]>>;
}

File: back-end/src/main/java/com/trucks/application/port/VesselRepository.java
package com.trucks.application.port;

import java.util.List;
import java.util.Optional;
import java.util.UUID;

import com.trucks.domain.entity.Vessel;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.PageRequest;

public interface VesselRepository {

    Vessel save(Vessel truck);

    Optional<Vessel> find(UUID id);

    Page<Vessel> findAll(PageRequest pageRequest);

    Vessel update(UUID id, Vessel truck);
    
    Vessel delete(UUID id);

    List<Vessel> deleteAll();
}
